src/components/formStore/table/useFormStoreTable.tsx

```diff
@@ -419,11 +422,13 @@ export default function useFormStoreTable({
   )
 
   React.useEffect(() => {
+    if ((table.state as FormTableState).formId !== form.id) {
```

Comment:
does the table.state eventually update to have the correct form id?

Reply:
yeh it calls the `initialState` callback again, once the formId changes